src/lib/QueryType.hs

```diff
@@ -374,6 +374,8 @@ instance HasType Atom where
                                      (Var v') -> return v'
                                      _ -> error "!?"
                                    instantiateDepPairTy t (ProjectElt (0 NE.:| is) v')
+        -- Newtypes
```

Comment:
I don't understand this comment.  Is it saying that TC (Fin _) is possible because of newtypes?  Also, should there be an arm for newtypes in this case statement?

Reply:
This is meant to query the type of a projection. Before this change, projections out of `Fin` were not allowed and you had to use a `CastOp`. Now, I allowed projections out of newtypes, and `Fin` is currently the only newtype we support.